py3status/modules/google_calendar.py

```diff
@@ -0,0 +1,502 @@
+                calendarId='primary', timeMin=now, maxResults=self.num_events, singleEvents=True,
+                orderBy='startTime').execute(num_retries=5)
+        except:
+            return event_cache
+
+        return eventsResult.get('items', [])
+
+    def _check_warn_threshold(self, time_to, start, end, summary):
+        """
+        Checks if the time until an event starts is less than or equal to the warn_threshold.
+        If it is, it presents a warning with self.py3.notify_user.
+        """
+        if time_to['total_minutes'] <= self.warn_threshold:
+            warn_message = str(summary) + " " + str(start) + " - " + str(end)
+            self.py3.notify_user(warn_message, 'warning', self.warn_timeout)
+
+    def _gstr_to_date(self, date_str):
+        """ Returns a dateime object from a google calendar date string."""
+        return parser.parse(date_str).replace(tzinfo=tzlocal())
+
+    def _gstr_to_datetime(self, date_time_str):
+        """ Returns a datetime object from a google calendar date/time string."""
+        return parser.parse(date_time_str)
+
+    def _datetime_to_str(self, date_time, dt_format):
+        """ Returns a strftime formatted string from a datetime object."""
+        return date_time.strftime(dt_format)
+
+    def _delta_time(self, date_time):
+        """
+        Returns in a dict the number of days/hours/minutes and total minutes
+        until date_time.
+        """
+        now = datetime.datetime.now(tzlocal())
+        diff = date_time - now
+
+        days = int(diff.days)
+        hours = int(diff.seconds / 3600)
+        minutes = int((diff.seconds / 60) - (hours * 60)) + 1
+        total_minutes = int((diff.seconds / 60) + (days * 24 * 60)) + 1
+
+        return {'days': days,
+                'hours': hours,
+                'minutes': minutes,
+                'total_minutes': total_minutes}
+
+    def _format_timedelta(self, index, time_delta, is_current):
+        """
+        Formats the dict time_to containg days/hours/minutes until an event starts
+        into a composite according to time_to_formatted.
+
+        Returns: A formatted composite.
+        """
+        time_delta_formatted = ''
+
+        if time_delta['total_minutes'] <= self.time_to_max:
+            time_delta_formatted = self.py3.safe_format(
+                self.format_timer,
+                {'days': time_delta['days'],
+                 'hours': time_delta['hours'],
+                 'minutes': time_delta['minutes'],
+                 'is_current': is_current})
+
+        return time_delta_formatted
+
+    def _build_response(self):
+        """
+        Builds the composite reponse to be output by the module by looping through all
+        events and formatting the necessary strings.
+
+        Returns: A composite containing the individual response for each event.
+        """
+        responses = []
+        self.event_urls = []
+        index = 0
+
+        for event in self.events:
```

Comment:
(Nit) `for index, event in enumerate(self.events):` will remove a line or two and is cleaner (imo).

Reply:
Neat. I'm still pretty new to Python, so I don't know a lot of these tricks.

Reply:
First time I use Python? Jan 2017. You're doing great. :+1: